pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
Yeah, admission controller seems like a natural choice for this. If the goal is to prevent any operation that can end up in granting or removing more privileges than the requester already has, that admission controller needs to work with all of RBAC resources on creates, updates, and deletes. I could be wrong though, as I only started looking at this change and might be missing some context.